cluster/kubemark/iks/config-default.sh

```diff
@@ -25,8 +25,13 @@ CLUSTER_LOCATION="${CLUSTER_LOCATION:-wdc06}"
 REGISTRY_LOGIN_URL="${REGISTRY_LOGIN_URL:-https://api.ng.bluemix.net}"
 
 # User defined
+# number of real workers in spawnTester cluster
```

Comment:
Nothing in kubemark should depend on NUM_NODES (as it shouldn't matter on how many nodes it has been run).

Instead, you should do the same that gce is doing:
NUM_NODES=${KUBEMARK_NUM_NODES:-10}
https://github.com/kubernetes/kubernetes/blob/master/cluster/kubemark/gce/config-default.sh#L29

Then you won't need a bunch of changes below too.

Reply:
Thanks for the prompt reivew! I read into the code, and noticed GCE works basically as below:

Firstly `--gcp-nodes` is passed to `kubetest`, and then it's passed onto env var:

```go
		{
			Env:    "NUM_NODES",
			Option: &o.gcpNodes,
			Name:   "--gcp-nodes",
		},
```

And then it uses bash (default) deployer to spin up a real cluster, which in turns calls `hack/e2e-internal/e2e-up.sh`:

```go
func (b *bashDeployer) Up() error {
	script := "./hack/e2e-internal/e2e-up.sh"
    ...
```

By this step, env var `NUM_NODES` literally means the number of the real cluster.

After that (the cluster is ready), `KUBEMARK_NUM_NODES` can overwrite the value of `NUM_NODES` (as the old value is not necessary any more), and we starts a hollow-node cluster as a rc.

```
NUM_NODES=${KUBEMARK_NUM_NODES:-10}
```

So basically GCE works in 2 phases: spin up a cluster and start the hollow-node rc. And they're working in serial, so we can overwrite NUM_NODES in latter phase with KUBEMARK_NUM_NODES.

For IKS, the embarrassing thing is `kubetest` doesn't support IKS yet, so `start-kubemark.sh` for IKS combines the 2 phases together. So IMO it's good to separate the 2 variables for IKS; otherwise it'd confuse people which literally means the number of the real cluster, and which means the number of the hollow-node rc.

Reply:
OK - I didn't know you don't have support in kubetest.
In such case, it's probably the easiest thing to do to unblock you.

Can you please add a TODO with a comment why you do that and to change that once kubetest supports IKS?

Reply:
That makes sense.

Updated. PTAL.